src/middlewares/authValidation.ts

```diff
@@ -1,32 +1,38 @@
-import { ValidationError } from '@/utils/custom-error';
-import { errorCode, roles } from '@/utils/utils';
-import { Context, Next } from 'hono';
-import userRepository from '@/repositories/userRepository';
-import { zValidator } from '@hono/zod-validator';
-import { SignSchema, UserBodySchema } from '@/validation/body-check';
+import { AuthError, ValidationError } from "@/utils/custom-error";
+import { errorCode, roles } from "@/utils/utils";
+import { Context, Next } from "hono";
+import userRepository from "@/repositories/userRepository";
+import { zValidator } from "@hono/zod-validator";
+import { SignSchema, UserBodySchema } from "@/validation/body-check";
 
 export const checkAdmin = async (c: Context, next: Next) => {
-  const payload = c.get('jwtPayload');
+  const payload = c.get("jwtPayload");
 
   const user = await userRepository.findByEmail(payload.email);
 
   if (user.role !== roles.ADMIN) {
-    throw new ValidationError('You are not allowed to this route', errorCode.INVALID);
+    throw new AuthError(
+      "You are not allowed to this route",
+      errorCode.FORBIDDEN
+    );
   }
   return next();
 };
 
 export const CheckUser = async (c: Context, next: Next) => {
```

Comment:
while we have only 2 roles, this function doesn’t make any sense because we can simple use jwt and it will be enough. But if we add a third role then this function is necessary